core/src/main/java/org/frankframework/util/flow/FlowDiagramManager.java

```diff
@@ -247,6 +245,28 @@ private void generateFlowDiagram(String name, String xml, File destination) thro
 		log.debug("finished generating flow diagram for [{}] in [{}] ms", () -> name, () -> System.currentTimeMillis() - start);
 	}
 
+	static String encodeFileName(String fileName) {
+		String mark = "-_.+=";
+		StringBuilder encodedFileName = new StringBuilder();
+		int len = fileName.length();
+		for (int i = 0; i < len; i++) {
+			char c = fileName.charAt(i);
+			if ((c >= '0' && c <= '9')
+					|| (c >= 'a' && c <= 'z')
+					|| (c >= 'A' && c <= 'Z'))
+				encodedFileName.append(c);
+			else {
+				int imark = mark.indexOf(c);
+				if (imark >= 0) {
+					encodedFileName.append(c);
+				} else {
+					encodedFileName.append('_');
+				}
+			}
+		}
+		return encodedFileName.toString();
+	}
```

Comment:
Beetje rare code. Die hele string `mark` lijkt compleet overbodig, zou ook kunnen met wat extra cases bij de `if`, of anders erbij ` || mark.indexOf(c) >= 0`.
Veel simpeler dan wat hier staat.

Reply:
Eens. En Commons helpt hier ons ook nog bij.